src/dstore/pmix_esh.c

```diff
@@ -58,6 +58,14 @@ pmix_dstore_base_module_t pmix_dstore_esh_module = {
 #define ESH_ENV_NS_DATA_SEG_SIZE    "NS_DATA_SEG_SIZE"
 #define ESH_ENV_LINEAR              "SM_USE_LINEAR_SEARCH"
 
+#define _ESH_LOCK(lockfd, operation) \
+do{ \
+    struct flock fl = {0}; \
+    fl.l_type = operation; \
+    fl.l_whence = SEEK_SET; \
+    fcntl(lockfd, F_SETLKW, &fl); \
```

Comment:
Please check the return value of the `fcntl` function. Coverity will flag this, and it is good to check anyway.